tests/core/test_field_types.py

```diff
@@ -71,11 +71,9 @@ def test_float_input(self):
         """ Test that floats and decimals are converted to integer"""
 
         age = field.Integer()
-        age.validate(3.1)
-        assert age.value == 3
+        assert age.validate(3.1) == 3
 
-        age.validate(Decimal(3.1))
-        assert age.value == 3
+        assert age.validate(Decimal(3.1)) == 3
```

Comment:
Looks like `validate` returns a value rather than returning `True` or `False`. This is a side-effect and mixing responsibilities, and should be avoided. Let Validation simply return whether the value is valid or not, based on the original value without transformation.

Sorry, I might be the one who caused this confusion. Let's keep to your thought process and not do ANY type conversions.